src/discussions/topics/data/slices.js

```diff
@@ -37,6 +40,34 @@ const topicsSlice = createSlice({
     },
     setFilter: (state, { payload }) => {
       state.filter = payload;
+      let count = 0;
+      const currentState = current(state);
+      // Counting non-courseware topics
+      const nonCoursewareTopicsList = currentState.nonCoursewareIds.map(
+        id => currentState.topics[id],
+      ).filter(item => (payload
+        ? item.name.toLowerCase().includes(payload)
+        : true
+      ));
+      count += nonCoursewareTopicsList.length;
+      // Counting legacy topics
+      const categories = currentState.categoryIds;
+      const filteredTopics = categories?.map(categoryId => {
+        const topics = currentState.topicsInCategory[categoryId]?.map(
+          id => currentState.topics[id],
+        ) || [];
+        const matchesFilter = payload ? categoryId?.toLowerCase().includes(payload) : true;
+        return topics.filter(
+          topic => (
+            payload
+              ? (topic.name.toLowerCase()
+                .includes(payload) || matchesFilter)
+              : true
+          ),
+        );
+      });
+      count += [].concat(...filteredTopics).length;
+      state.results.count = count;
```

Comment:
These array operations will affect the redux mutability rule. Please perform any calculation before dispatching an action. The redux store's only purpose is to store data. 
cc: @mehaknasir